site/reference/fn/eval/README.md

```diff
@@ -0,0 +1,153 @@
+---
+title: "`eval`"
+linkTitle: "eval"
+type: docs
+description: >
+  Execute function on resources
+---
+
+<!--mdtogo:Short
+    Execute function on resources
+-->
+
+`eval` executes a function on resources in a directory. Functions are packaged
+as container images or binary executables.
+
+If the function fails (i.e. exits with non-zero status code), `eval` will
+abort and the local filesystem is left intact.
+
+Refer to the [Imperative Function Execution] for detailed overview.
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```shell
+kpt fn eval [DIR|-] [flags] [-- fn-args]
+```
+
+#### Args
+
+```shell
+DIR|-:
+  Path to the local directory containing resources. Defaults to the current
+  working directory. Using '-' as the directory path will cause `eval` to
+  read resources from `stdin` and write the output to `stdout`.
```

Comment:
When using Unix pipe:
https://kpt.dev/book/04-using-functions/02-imperative-function-execution?id=chaining-functions-using-the-unix-pipe

The human-readable presentation should be sent to to `stderr` right? Currently, this doesn't display anything.

Reply:
we are completely suppressing the human-readable presentation in this case which is not good. `stderr` could be an option, but should that be default ?

Reply:
what's the alternative? How unix pipes work is that by default, stdout is piped and stderr is used for observability.

Reply:
ok, then we can use stderr.